woocommerce/woocommerce-gutenberg-products-block
# Check if the current theme has a product-archive block template in the absence of a product taxonomy block template.

This change means that if a theme has a `product-archive.html` block template, but not more taxonomy specific ones we will use the `archive-product.html` template for taxonomy frontend _instead of_ the Woo Blocks `taxonomy-product_cat`/`taxonomy-product_tag` files.


Fixes https://github.com/woocommerce/woocommerce-gutenberg-products-block/issues/5230

### Testing

### Manual Testing

How to test the changes in this Pull Request:

To make this easier to test, add the following block markup to the following templates so you know which one is getting rendered:

```

<p>Woo Blocks: [template-filename].html</p>

```

* `woo-blocks/archive-product.html`
* `woo-blocks/taxonomy-product_tag.html`
* `woo-blocks/taxonomy-product_cat.html`

1. Install Gutenberg plugin and Blockbase theme to activate Site Editor and FSE.
2. Go to a product category page and check that `woo-blocks/taxonomy-product_cat.html` is the template being used.
3. Duplicate `woo-blocks/archive-product.html` into `blockbase/block-templates/` and change the markup you added above to say "Blockbase" instead of "Woo Blocks" so we can differentiate.
4. Reload the product category page, product tag page, and main shop page/product archive page and check that the `archive-product.html` from Blockbase is being loaded.
5. Go into Site Editor > All Templates and check that "Product Archive Page", "Product Category Page" and "Product Tag Page" are all in the list and it states they're being loaded from the Blockbase theme
6. Customise the Product Category Page template and save your changes
7. Reload the product category page on the frontend and check the customised version of this template is being rendered.

### User Facing Testing
These are steps for user testing (where "user" is someone interacting with this change that is not editing any code).

1.
2.
3.

### Changelog

> Use the themes product archive block template for product category & product tag pages if the theme does not have more specific templates for those.
